src/main/java/io/localmotion/chatbox/notification/ChatBoxNotificationRepository.java

```diff
@@ -0,0 +1,70 @@
+package io.localmotion.chatbox.notification;
+
+import io.micronaut.configuration.hibernate.jpa.scope.CurrentSession;
+import io.micronaut.spring.tx.annotation.Transactional;
+import org.springframework.transaction.annotation.Propagation;
+
+import javax.inject.Singleton;
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+
+@Singleton
+public class ChatBoxNotificationRepository implements IChatBoxNotificationRepository {
+
+	@PersistenceContext
+	private EntityManager entityManager;
+
+	public ChatBoxNotificationRepository(@CurrentSession EntityManager entityManager) {
+		this.entityManager = entityManager;
+	}
+
+	public static final String selectDistinctUsers = "select distinct m.chatBoxNotificationId.personId from ChatBoxNotification m ";
```

Comment:
As a general pattern I propose that the query identifier is <methodname>Query
So if the method is `findDistinctPersons` then the query should be called `findDistinctPersonsQuery`

Reply:
Since `selectDistinctUsers` is a constant. can I name it as `FIND_DISTINCT_PERSONS_QUERY`?